docs/reference/files/charmcraft-yaml-file.rst

```diff
@@ -424,6 +432,46 @@ secret URI.
         :start-at: config:
         :end-before: containers:
 
+.. admonition:: Best Practice
+    :class: hint
+
+    Charms should keep the user experience of the operator as simple and obvious
+    as possible. Aim to make the application deployable without providing any
+    further configuration options: ``juju deploy foo`` is preferable to
+    ``juju deploy foo --config required-option=value``.
+
+.. admonition:: Best Practice
+    :class: hint
+
+    Consider which of your application's configuration options you should
+    initially expose. If your application has many config options, it may be
+    prudent to provide access to a select few, and add support for more as the
+    need arises.
+
+.. admonition:: Best Practice
+    :class: hint
+
+    Prefer lowercase alphanumeric names, separated with dashes if required. For
+    charms that have already standardised on underscores, it is not necessary to
+    change them, and it is better to be consistent within a charm then to have
+    some config names be dashed and some be underscored.
+
+.. admonition:: Best Practice
+    :class: hint
+
+    The description a string type. YAML supports two types of formats for that:
+    ``block scalar`` and ``flow scalar``. Prefer the block style (using ``|``)
+    to keep newlines. Using ``>`` will replace newlines with spaces and make
+    the result harder to read on Charmhub.
+
+.. tip::
+
+  For very complex applications, consider providing "configuration profiles",
+  which can group values for large configs together. For example,
+  "profile: large" that tweaks multiple options under the hood to optimise for
+  larger deployments, or "profile: ci" for limited resource usage during
+  testing.
+
 
```

Comment:
@lengau Do we really want this?

Reply:
I think this is fine to keep, though it's probably a best practice as well.